src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -133,13 +137,22 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext executionCon
                     throw new ArgumentException($"Input path '{path}' should resolve to a directory under '{buildDirectory}', current resolved path '{expectRepoPath}'.");
                 }
             }
+            else if (HasMultipleCheckouts(executionContext))
+            {
+                // When repository doesn't has path set, default to directory 1/<repoName>
+                expectRepoPath = Path.Combine(buildDirectory, RepositoryUtil.GetCloneDirectory(repo));
+            }
             else
             {
-                // When repository doesn't has path set, default to sources directory 1/s
+                // When there's a single checkout task that doesn't has path set, default to sources directory 1/s
                 expectRepoPath = Path.Combine(buildDirectory, "s");
             }
 
-            executionContext.UpdateRepositoryPath(repoAlias, expectRepoPath);
+            // Only update the repo path variables if there is a single checkout task
+            if (!HasMultipleCheckouts(executionContext))
```

Comment:
Nit: Can we combine this with the logic above? The outer if can check if there are multiple checkouts. Reads a bit strange to check in the else if then check again here. It seems less likely to have edge case bugs that way.

Also ties to my question above about path being null.

Reply:
I will see what I can do.

Reply:
I actually had to change this a bit to fix a problem. So, it should read a bit better now, but the location of the code is still basically the same.